### PR TITLE
Replace createCore by Core constructor. Move headers to include directory. Add some helpers function to lib.

### PR DESCRIPTION
**Pull request by artyomtugaryov**

### CODE REVIEW

File: example/hello_classification_node/main.js [hunk 1]
```diff
@@ -291,6 +247,7 @@ async function main() {
     } else {
       input_data = new Float32Array(input_blob.wmap());
```
Comment (artyomtugaryov) [inline example/hello_classification_node/main.js:250]: Why do we read as `Uint8Array` if not preprocess?

**Comment by artyomtugaryov** [comment]
@huningxin @lionkunonly What do you think about moving headers to the separated directory `include`?

File: example/hello_classification_node/main.js [hunk 2]
```diff
@@ -195,92 +144,99 @@ async function main() {
...
+  } else if (std.length !== 3) {
     warning('The length of std is not 3.');
     process.exit(-1);
   }
-  const preprocess = (mean[0] === 0 && mean[1] === 0 && mean[2] === 0 &&
-                      std[0] === 1 && std[1] === 1 && std[2] === 1) ?
-      false :
-      true;
+  const preprocess = !(mean[0] === 0 && mean[1] === 0 && mean[2] === 0 &&
+                       std[0]  === 1 && std[1]  === 1 && std[2]  === 1);
```
Comment (huningxin) [inline example/hello_classification_node/main.js]: Why add another space before `===`?

File: example/hello_classification_node/main.js (hunk 1 above)
Comment (huningxin) [reply example/hello_classification_node/main.js:250]: We set `input_info.setPrecision('u8');` if not preprocess. We do preprocess in JS currently. I think we should use `PreProcessInfo` that is supported with #30 merged.

**Review by huningxin — APPROVED** [review]
Looks good with a nit. Thanks much @artyomtugaryov !

File: src/blob.cc [hunk 3]
```diff
@@ -17,14 +17,14 @@ Napi::FunctionReference Blob::constructor;
...
-       InstanceMethod("wmap", &Blob::Wmap),
-       InstanceMethod("unmap", &Blob::Unmap)});
+  Napi::Function func = DefineClass(env, "Blob", {
+      InstanceMethod("byteSize", &Blob::ByteSize),
+      InstanceMethod("size", &Blob::Size),
+      InstanceMethod("rmap", &Blob::Rmap),
+      InstanceMethod("rwmap", &Blob::Rwmap),
+      InstanceMethod("wmap", &Blob::Wmap),
+      InstanceMethod("unmap", &Blob::Unmap)
+  });
```
Comment (lionkunonly) [inline src/blob.cc]: Do you use any script to get such a code style? You can share it so that we can keep the same code style.

Comment (artyomtugaryov) [reply src/blob.cc]: Actually, I use auto formating in CLion. But it is a good idea to set up some tools to lint code. I will try it. 
As I understand you don't have CI for this repo. Am I right?

File: example/hello_classification_node/main.js (hunk 2 above)
Comment (artyomtugaryov) [reply example/hello_classification_node/main.js]: just to align mean and std lines. I can remove these spaces

File: src/blob.cc (hunk 3 above)
Comment (huningxin) [reply src/blob.cc]: We have coding style doc: https://github.com/intel/inference-engine-node/blob/master/doc/coding_style.md


> But it is a good idea to set up some tools to lint code. I will try it.

 +1. Thanks in advance!

**Comment by huningxin** [comment]
> What do you think about moving headers to the separated directory `include`?

It sounds good to me.

File: src/blob.cc (hunk 3 above)
Comment (artyomtugaryov) [reply src/blob.cc]: Ok, I will try this tool

File: example/hello_classification_node/main.js (hunk 1 above)
Comment (artyomtugaryov) [reply example/hello_classification_node/main.js:250]: I think I can use `PreProcessInfo` in the next PRs

File: example/hello_classification_node/main.js (hunk 2 above)
Comment (artyomtugaryov) [reply example/hello_classification_node/main.js]: Fixed after running `clang-format -i -style=Google `

File: src/blob.cc (hunk 3 above)
Comment (artyomtugaryov) [reply src/blob.cc]: Fixed all styles after running `clang-format`. I suggest adding auto-formatting using clang-format to git hooks. For example, `clang-format` will be automatically run when we run `git add file` command

**Comment by artyomtugaryov** [comment]
@huningxin can we merge it?

**Comment by huningxin** [comment]
Sure, it is in good shape now. Thanks much this improvement, @artyomtugaryov ! Let me merge it.

**Pull request merged**